database/api.go

```diff
@@ -0,0 +1,174 @@
+//go:generate mockgen -source=$GOFILE -package $GOPACKAGE -destination api.mock.gen.go
+package database
+
+import (
+	"github.com/beaconsoftwarellc/gadget/v2/database/qb"
+	"github.com/jmoiron/sqlx"
+)
+
+// API is a database interface
+type API interface {
```

Comment:
We could do a major version change and make `Database->database` and rename this to `Database`

Reply:
`"github.com/beaconsoftwarellc/gadget/database/deltas"` depends on `Database` locks. We can move it to the interface as well.